### DISCUSSION TITLE
Is MIR a good fit for an AOT statically & strongly typed language with manual memory management?

### DISCUSSION THREAD

**Discussion by rishavs** [vnmakarov/mir#334] [General]
I am making a toy language and looking for a backend to host it.
two of my key criteria are;
1. Ease of integration
2. compilation speed

While binary execution speed is important, I will be happy if I can get 80% of the way of gcc/LLVMfor this.

Would MIR be a good fit for my language frontend? MIR seems to be heavily focused on JIT, which I don't plan on using.

Also, is there a Discord or any other channels for real time discussions on MIR?

**Comment by vnmakarov** [comment]
AOT is currently out of scope of the project as it requires generation of some loader format, like elf or/and macho for macosx .  MIR currently generates machine code  directly in memory.

TinyC generates code faster and generate object files.  But the generated code performance is about 50% of GCC/LLVM.  You can find diffent C compiler comaprison in one my blogpost https://developers.redhat.com/blog/2021/04/27/the-mir-c-interpreter-and-just-in-time-jit-compiler#how_the_mir_c_compiler_compares_with_other_c_compilers

Besides this I'd say MIR-project meets your goals.  You can achieve 80% code performance of GCC/LLVM by much faster JIT compiler (saying 30-60 times faster),

You can generate MIR-code directly or C code which can be translated into MIR and then machine code by `c2m` compiler which is a part of `libmir` library,  Using C probably will makes JIT compiler speed 40% slower.

MIR-project is mostly one-person project.  Although some people sends pull-requests from time to time and provides bug reports and tests and valuable feedback.  So I can not proved prompt answers and be on discord and any other channel.  I already spend much my spare time on this project.

**Comment by rishavs** [comment]
Thank you for the clarification.
